src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class VersionConverter : JsonConverter<Version>
+    {
+        public override Version? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            string? versionString = reader.GetString();
+            if (Version.TryParse(versionString, out Version? result))
+            {
+                return result;
+            }
+
+            ThrowHelper.ThrowJsonException();
```

Comment:
I think throwing like this is fine (where we have raw JsonException without special text for Version and without inner exception from Parse). So as-is we'll get a fairly nice message like this:
```
System.Text.Json.JsonException : The JSON value could not be converted to System.Version. Path: $ | LineNumber: 0 | BytePositionInLine: 2.
```

Reply:
Yup we have the same pattern for [Uri](https://github.com/dotnet/runtime/blob/98cba4cec14274fd2abaa88534814eac1e7eb573/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/UriConverter.cs#L16) and [KeyValuePair](https://github.com/dotnet/runtime/blob/98cba4cec14274fd2abaa88534814eac1e7eb573/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/KeyValuePairConverter.cs#L81).